app/message-table.tsx

```diff
@@ -17,19 +20,65 @@ import {
   TableHeader,
   TableRow,
 } from "@/components/ui/table";
-
-import { useState } from "react";
+import { Checkbox } from "@/components/ui/checkbox";
 
 interface MessageTableProps<TData, TValue> {
   columns: ColumnDef<TData, TValue>[];
   data: TData[];
   defaultExpanded?: boolean;
+  canHideMessages?: boolean;
+  impressionsThreshold?: string;
+}
+
+function MessageTableHead(
+  header: any,
+  hideMessages: boolean,
+  setHideMessages: any,
```

Comment:
Does this want to be a function type?

Reply:
I left it as a function type because of the arguments I needed to pass into it